homeassistant/components/home_connect/light.py

```diff
@@ -0,0 +1,207 @@
+                        self._colorkey,
+                        BSH_AMBIENT_LIGHT_COLOR_CUSTOM_COLOR,
+                    )
+                except HomeConnectError as err:
+                    _LOGGER.error("Error while trying selecting customcolor: %s", err)
+                if self._brightness is not None:
+                    brightness = 10 + ceil(self._brightness / 255 * 90)
+                    if ATTR_BRIGHTNESS in kwargs:
+                        brightness = 10 + ceil(kwargs[ATTR_BRIGHTNESS] / 255 * 90)
+
+                    hs_color = kwargs.get(ATTR_HS_COLOR, default=self._hs_color)
+
+                    if hs_color is not None:
+                        rgb = color_util.color_hsv_to_RGB(*hs_color, brightness)
+                        hex_val = color_util.color_rgb_to_hex(rgb[0], rgb[1], rgb[2])
+                        try:
+                            await self.hass.async_add_executor_job(
+                                self.device.appliance.set_setting,
+                                self._customcolorkey,
+                                f"#{hex_val}",
+                            )
+                        except HomeConnectError as err:
+                            _LOGGER.error(
+                                "Error while trying setting the color: %s", err
+                            )
+            else:
+                _LOGGER.debug("Tried to switch ambient light on for: %s", self.name)
+                try:
+                    await self.hass.async_add_executor_job(
+                        self.device.appliance.set_setting,
+                        self._key,
+                        True,
+                    )
+                except HomeConnectError as err:
+                    _LOGGER.error(
+                        "Error while trying to turn on ambient light: %s", err
+                    )
+                    self._state = False
+
+        elif ATTR_BRIGHTNESS in kwargs:
+            _LOGGER.debug("Tried to change brightness for: %s", self.name)
+            brightness = 10 + ceil(kwargs[ATTR_BRIGHTNESS] / 255 * 90)
+            try:
+                await self.hass.async_add_executor_job(
+                    self.device.appliance.set_setting,
+                    self._brightnesskey,
+                    brightness,
+                )
+            except HomeConnectError as err:
+                _LOGGER.error("Error while trying set the brightness: %s", err)
+                self._brightness = None
+        else:
+            _LOGGER.debug("Tried to switch light on for: %s", self.name)
+            try:
+                await self.hass.async_add_executor_job(
+                    self.device.appliance.set_setting,
+                    self._key,
+                    True,
+                )
+            except HomeConnectError as err:
+                _LOGGER.error("Error while trying to turn on light: %s", err)
+                self._state = False
```

Comment:
We shouldn't change state on error. Possibly set the entity to unavailable if that's what this error means.

Reply:
For consistency reason I followed the same solution as done in the platform switch. 
See switch.py L101

`    async def async_turn_on(self, **kwargs):
        """Switch the device on."""
        _LOGGER.debug("Tried to switch on %s", self.name)
        try:
            await self.hass.async_add_executor_job(
                self.device.appliance.set_setting,
                BSH_POWER_STATE,
                BSH_POWER_ON,
            )
        except HomeConnectError as err:
            _LOGGER.error("Error while trying to turn on device: %s", err)
            self._state = False
        self.async_entity_update()`

Reply:
We shouldn't do that. We don't know that that's the correct thing to do.

Reply:
Bump

Reply:
removed state setting after error